libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -808,7 +817,23 @@ bool AP_Arming::mission_checks(bool report)
                 check_failed(ARMING_CHECK_MISSION, report, "No sufficently close rally point located");
                 return false;
             }
-          }
+        }
+        if (_required_mission_items & MIS_ITEM_CHECK_NAV_DIST) {
+            AP_Mission::Mission_Command cmd;
+            Location ahrs_loc;
+            if (!AP::ahrs().get_location(ahrs_loc)) {
+                check_failed(ARMING_CHECK_MISSION, report, "Can't check distance to waypoint without position");
+                return false;
+            }
+            if (mission->get_next_nav_cmd_with_loc(cmd)) {
+                uint32_t max_dist = get_max_arm_dist_m();
+                float dist_to_wp = ahrs_loc.get_distance(cmd.content.location);
+                if (dist_to_wp > max_dist && max_dist) {
+                    check_failed(ARMING_CHECK_MISSION, report, "Check distance to first waypoint");
```

Comment:
I think this message usefull, maybe "First way point more than Xm from current location"